### PR TITLE
Development/stringpool api

### PR DESCRIPTION
**Pull request by Dyronix**

### CODE REVIEW

File: source/1_engine/rex_engine/src/string/stringpool.cpp [hunk 1]
```diff
@@ -52,25 +52,25 @@ namespace rex
       auto result = get_entries().emplace(rsl::move(entry_id), rsl::move(entry));
       if(result.emplace_successful)
       {
-        return &result.inserted_element->key;
+        return result.inserted_element->key;
       }
 
       REX_ASSERT("This path should never be reached, insertion into the string pool failed somehow.");
-      return nullptr;
+      return {};
```
Comment (RisingLiberty) [inline source/1_engine/rex_engine/src/string/stringpool.cpp:59]: can this be named instead or return StringEntryID(invalid_something_something);
eg. a global constexpr invalid stringid (like std::optional has with std::nullopt)

File: source/1_engine/rex_engine/src/string/stringpool.cpp [hunk 2]
```diff
@@ -80,11 +80,11 @@ namespace rex
         REX_ASSERT_X(it != rsl::cend(get_entries()), "StringID::is_none() not present");
       }
 
-      return &(it->value);
+      return (it->value);
```
Comment (RisingLiberty) [inline source/1_engine/rex_engine/src/string/stringpool.cpp:83]: unnecessary brackets

**Review by RisingLiberty — CHANGES_REQUESTED** [review]

File: source/1_engine/rex_engine/src/string/stringpool.cpp (hunk 1 above)
Comment (Dyronix) [reply source/1_engine/rex_engine/src/string/stringpool.cpp:59]: This has to become an invalid_object but we don't have the code available on this branch.

**Review by RisingLiberty — CHANGES_REQUESTED** [review]
Please resolve the merge conflicts

**Review by RisingLiberty — APPROVED** [review]

**Pull request merged**